python/lsst/sims/catUtils/utils/avroAlertGenerator.py

```diff
@@ -0,0 +1,324 @@
+
+        log_file_name is the name of an optional text file to which progress is
+        written.
+        """
+
+
+        out_name = os.path.join(out_dir,'%s_%d.avro' % (out_prefix, obshistid))
+        if os.path.exists(out_name):
+            os.unlink(out_name)
+
+        with DataFileWriter(open(out_name, "wb"),
+                            DatumWriter(), self._alert_schema) as data_writer:
+
+            diasource_query = 'SELECT alert.uniqueId, alert.xPix, alert.yPix, '
+            diasource_query += 'alert.chipNum, alert.dflux, alert.snr, alert.ra, alert.dec, '
+            diasource_query += 'meta.band, meta.TAI, quiescent.flux, quiescent.snr '
+            diasource_query += 'FROM alert_data as alert '
+            diasource_query += 'INNER JOIN metadata AS meta ON alert.obshistId=meta.obshistId '
+            diasource_query += 'INNER JOIN quiescent_flux AS quiescent ON quiescent.uniqueId=alert.uniqueID '
+            diasource_query += 'AND quiescent.band=meta.band '
+            diasource_query += 'WHERE alert.obshistId=%d ' % obshistid
+            diasource_query += 'ORDER BY alert.uniqueId'
+
+            diasource_dtype = np.dtype([('uniqueId', int), ('xPix', float), ('yPix', float),
+                                        ('chipNum', int), ('dflux', float), ('tot_snr', float),
+                                        ('ra', float), ('dec', float), ('band', int), ('TAI', float),
+                                        ('quiescent_flux', float), ('quiescent_snr', float)])
+
+            diaobject_query = 'SELECT uniqueId, ra, dec, TAI, pmRA, pmDec, parallax '
+            diaobject_query += 'FROM baseline_astrometry'
+
+            diaobject_dtype = np.dtype([('uniqueId', int), ('ra', float), ('dec', float),
+                                        ('TAI', float), ('pmRA', float), ('pmDec', float),
+                                        ('parallax', float)])
+
+            t_start = time.time()
+            alert_ct = 0
+            for htmid in htmid_list:
+                for prefix in prefix_list:
+                    db_name = os.path.join(data_dir, '%s_%d_sqlite.db' % (prefix, htmid))
+                    if not os.path.exists(db_name):
+                        warnings.warn('%s does not exist' % db_name)
+                        continue
+
+                    db_obj = DBObject(db_name, driver='sqlite')
+
+                    diaobject_data = db_obj.execute_arbitrary(diaobject_query,
+                                                              dtype=diaobject_dtype)
+
+                    diaobject_dict = self._create_objects(diaobject_data)
+
+                    diasource_data = db_obj.execute_arbitrary(diasource_query,
+                                                              dtype=diasource_dtype)
+
+                    dmag = 2.5*np.log10(1.0+diasource_data['dflux']/diasource_data['quiescent_flux'])
+                    valid_alerts = np.where(np.abs(dmag)>=dmag_cutoff)
```

Comment:
I'm okay with it.  There are a lot of moving parts to this, so people are going to have to be careful.  Also: people will presumably use `AlertDataGenerator` for more than just generating alert streams (for example: generating a small subset of very detailed light curves).

That being said, we could probably add a metadata table to each sqlite file explaining the parameters with which it was generated.  Do you mind if I punt that down to another PR?

Reply:
That sounds good. I was kind of wondering if there was a way to include that info, but wasn't sure if it would make things more complicated than it was worth.